code/_onclick/hud/parallax.dm

```diff
@@ -126,29 +126,31 @@ var/list/parallax_on_clients = list()
 		C.previous_turf = posobj
 
 	//Doing it this way prevents parallax layers from "jumping" when you change Z-Levels.
-	C.parallax_offset["horizontal"] += posobj.x - C.previous_turf.x
-	C.parallax_offset["vertical"] += posobj.y - C.previous_turf.y
+	var/offsetx = C.parallax_offset["horizontal"] + posobj.x - C.previous_turf.x
+	var/offsety = C.parallax_offset["vertical"] + posobj.y - C.previous_turf.y
+	C.parallax_offset["horizontal"] = offsetx
+	C.parallax_offset["vertical"] = offsety
 
 	C.previous_turf = posobj
 
+	var/maxoffset = 480 //480 = (15 tiles * 32 icon_size * 3 grid size / 2) - (15 tiles * 32 icon size / 2) for centering
+	var/minoffset = -960 //960 = (15 tiles * 32 icon_size * 3 grid size / 2) + (15 tiles * 32 icon size / 2) for centering
+
 	for(var/obj/screen/parallax/bgobj in C.parallax_movable)
-		if(bgobj.parallax_speed)//only the middle and front layers actually move
-			var/accumulated_offset_x = bgobj.base_offset_x - round(C.parallax_offset["horizontal"] * bgobj.parallax_speed * (C.prefs.parallax_speed/2))
-			var/accumulated_offset_y = bgobj.base_offset_y - round(C.parallax_offset["vertical"] * bgobj.parallax_speed * (C.prefs.parallax_speed/2))
+		var/accumulated_offset_x = bgobj.base_offset_x - round(offsetx * bgobj.parallax_speed * C.prefs.parallax_speed)
+		var/accumulated_offset_y = bgobj.base_offset_y - round(offsety * bgobj.parallax_speed * C.prefs.parallax_speed)
 
-			while(accumulated_offset_x > 720)
-				accumulated_offset_x -= 1440
-			while(accumulated_offset_x < -720)
-				accumulated_offset_x += 1440
+		if(accumulated_offset_x > maxoffset)
+			accumulated_offset_x -= 1440 //3x3 grid, 15 tiles * 32 icon_size * 3 grid size
```

Comment:
It's not a compile time value and therefore it will be calculated, a LOT. If it wasn't the primary update loop I would have done it that way.